airbyte-cdk/python/unit_tests/sources/file_based/scenarios/_base_scenario.py

```diff
@@ -0,0 +1,37 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Dict, Mapping
+
+from unit_tests.sources.file_based.in_memory_files_source import InMemoryFilesSource
+
+
+class BaseTestScenario:
+    name: str
+    config: Mapping[str, Any]
+    files: Dict[str, Any]
+    expected_catalog: Dict[str, Any]
+    expected_records: Dict[str, Any]
+
+    def __init__(self):
+        self.source = InMemoryFilesSource(self.files)
+        self.validate()
+
+    def validate(self):
+        streams = {s["name"] for s in self.config["streams"]}
+        expected_streams = {s["name"] for s in self.expected_catalog["streams"]}
+        assert expected_streams <= streams
```

Comment:
n00b question since I haven't run this myself: if a class implementing this base scenario fails, is it easy to know exactly which test case failed? e.g: when we do `pytest.mark.parametrize` you can give individual names to test cases and have individualized error messages and an ❌ shows on the exact test case which failed etc.. which I think is a useful thing. Is that also the case here or do all these inputs show up as one big test?

Reply:
As long as the scenario is instantiated as part of a test function (either a `pytest.mark.parametrize` of a stand-alone), I don't see this problem occurring